module/zfs/zio.c

```diff
@@ -2584,17 +2584,24 @@ int
 zio_resume(spa_t *spa)
 {
 	zio_t *pio;
+	zio_suspend_reason_t reason;
 
 	/*
 	 * Reexecute all previously suspended i/o.
 	 */
 	mutex_enter(&spa->spa_suspend_lock);
+	reason = spa->spa_suspended;
 	spa->spa_suspended = ZIO_SUSPEND_NONE;
 	cv_broadcast(&spa->spa_suspend_cv);
 	pio = spa->spa_suspend_zio_root;
 	spa->spa_suspend_zio_root = NULL;
 	mutex_exit(&spa->spa_suspend_lock);
 
+	if (reason != ZIO_SUSPEND_NONE)
+		cmn_err(CE_WARN, "Pool '%s' was suspended and is being "
+		    "resumed. Failed I/O will be retried.",
+		    spa_name(spa));
```

Comment:
I suppose your intention was to move the logging out of the lock, which is nice, but this way I guess some activity might happen on the `cv_broadcast()` before the message is printed, that might be confusing.  Is there a strong reason not to call `cmn_err()` under the lock?  I doubt lock contention matters here.

Reply:
Yeah, you're right, no particular reason to do it this way. Changed as you suggest, thanks.